### PR TITLE
Add compile-time unrolled for loops

### PR DESCRIPTION
**Pull request by juancarlospaco**
- Add compile-time unrolled for loops.
- Documentation, examples, links, runnableExamples, changelog, etc.
- Tiny diff.

### CODE REVIEW

File: lib/system/iterators_1.nim [hunk 1]
```diff
@@ -178,7 +180,7 @@ else: # not defined(nimNewRoof)
       inc i
 
 
-iterator `||`*[S, T](a: S, b: T, annotation: static string = "parallel for"): T {.
+iterator `||`*[S, T](a: S, b: T, annotation: static string = "omp parallel for"): T {.
```
Comment (alaviss) [inline lib/system/iterators_1.nim:181]: If you need a new iterator type, make one, don't abuse the parallel for iterator.

Removing the `omp` as a prefix of the iterator annotation is a breaking change for anyone using this iterator with a custom annotation.

**Review by alaviss — CHANGES_REQUESTED** [review]

File: lib/system/iterators_1.nim (hunk 1 above)
Comment (juancarlospaco) [reply lib/system/iterators_1.nim:181]: I dont understand what you mean, the `||` keeps working.

Comment (alaviss) [reply lib/system/iterators_1.nim:181]: Now anyone that use `||` with a custom annotation must prefix their annotation with `omp`, so this is a breaking change.

@mratsim's laser is [a big user of this iterator](https://github.com/numforge/laser/blob/master/laser/openmp.nim), and this change will cause it not to compile.

Comment (alaviss) [reply lib/system/iterators_1.nim:181]: My advise would be to create a brand new loop type. And note that `#pragma unroll` can unroll any loop, so a solution that allows tagging a generated loop would be much more efficient. Basically implementing the `{.unroll.}` pragma, if you may.

**Pull request closed without merging**